src/main/java/de/gerrygames/viarewind/legacysupport/injector/BoundingBoxFixer.java

```diff
@@ -45,28 +47,63 @@ public static void fixLadder() {
-		if (field.getType().getSimpleName().equals("AxisAlignedBB")) {
-			Class boundingBoxClass = field.getType();
-			Constructor boundingBoxConstructor = boundingBoxClass.getConstructor(double.class, double.class, double.class, double.class, double.class, double.class);
-			Object boundingBox = boundingBoxConstructor.newInstance(x1, y1, z1, x2, y2, z2);
-			ReflectionAPI.setFinalValue(field, boundingBox);
-		} else if (field.getType().getSimpleName().equals("VoxelShape")) {
-			Method createVoxelShape = ReflectionAPI.getMethod(NMSReflection.getNMSClass("VoxelShapes"), "create", double.class, double.class, double.class, double.class, double.class, double.class);
-			Object boundingBox = ReflectionAPI.invokeMethod(createVoxelShape, x1, y1, z1, x2, y2, z2);
-			ReflectionAPI.setFinalValue(field, boundingBox);
-		} else {
-			throw new IllegalStateException();
+	private static void setBoundingBox(Object boundingBox, double... values) throws ReflectiveOperationException {
+		if (boundingBox.getClass().getSimpleName().equals("VoxelShapeArray")) {
+			setVoxelShapeArray(boundingBox, values);
+			return;
+		}
+
+		if (boundingBox.getClass().getSimpleName().equals("AxisAlignedBB")) {
+			setAxisAlignedBB(boundingBox, values);
+			return;
+		}
+
+		throw new IllegalStateException("Unknown bounding box type: " + boundingBox.getClass().getName());
+	}
+
+	private static void setAxisAlignedBB(Object boundingBox, double[] values) throws ReflectiveOperationException {
+		Field[] doubleFields = Arrays.stream(boundingBox.getClass().getDeclaredFields())
+				.filter(f -> f.getType() == double.class)
+				.toArray(Field[]::new);
+
+		if (doubleFields.length < 6) {
+			throw new IllegalStateException("Invalid field count for " + boundingBox.getClass().getName() + ": " + doubleFields.length);
+		}
+
+		for (int i = 0; i < 6; i++) {
+			Field currentField = doubleFields[i];
+			currentField.setAccessible(true);
+			currentField.setDouble(boundingBox, values[i]);
+		}
+	}
+
+	private static void setVoxelShapeArray(Object voxelShapeArray, double[] values) throws ReflectiveOperationException {
+		Field[] doubleListFields = Arrays.stream(voxelShapeArray.getClass().getDeclaredFields())
+				.filter(f -> f.getType().getSimpleName().equals("DoubleList"))
+				.toArray(Field[]::new);
+
+		if (doubleListFields.length < 3) {
+			throw new IllegalStateException("Invalid field count for " + voxelShapeArray.getClass().getName() + ": " + doubleListFields.length);
+		}
+
+		// fastutil is relocated on Spigot but not on Paper
+		String doubleArrayListClass = doubleListFields[0].getType().getName().replace("DoubleList", "DoubleArrayList");
+		Method wrapMethod = Class.forName(doubleArrayListClass).getMethod("wrap", double[].class);
+
+		for (int i = 0; i < 3; i++) {
+			double[] array = {values[i * 2], values[i * 2 + 1]};
```

Comment:
Should this be `{values[i], values[i+3]}` instead? 
As it stands, this puts `xmin, ymin` `zmin xmax` `ymax zmax` into the 3 `DoubleList`s.

Reply:
Thanks, it's fixed